karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/replace.feature

```diff
@@ -63,4 +66,155 @@ Feature: Patch patient - Replace data
+      * def commLanguageDls = response.extension.find(x => x.url == commExtensionUrl)
+      * def interpreterDls = commLanguageDls[1]
+      * def commLanguageindex = response.extension.findIndex(x => x.url == commExtensionUrl)
+      * def interpreterPath = "/extension/" + commLanguageindex + "/extension/1"
+      * def interpreter = response.extension[commLanguageindex].extension[1].valueBoolean
+      * def oppositeInterpreter = karate.eval('function getOppositeBoolean(value) { return !value; } getOppositeBoolean(' + karate.toString(interpreter) + ')')
+     
+    # update comminication language interpreter required details
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * header Content-Type = "application/json-patch+json"
+      * header If-Match = karate.response.header('etag')
+      * path 'Patient', nhsNumber
+      * request 
+      """
+      {
+        "patches":[
+          {"op":"replace",
+          "path":"#(interpreterPath)",
+          "value": {
+                    "url": "interpreterRequired",
+                    "valueBoolean": "#(oppositeInterpreter)"              
+                   } 
+          }]}
+       """       
+      * method patch
+      * status 200
+      * match parseInt(response.meta.versionId) == originalVersion + 1
+  
+    Scenario: Send empty field on the update - interpreterRequired url is empty
+      * def nhsNumber = '5900071413'
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * path 'Patient', nhsNumber
+      * method get
+      * status 200
+      * def originalVersion = parseInt(response.meta.versionId)
+      * def commExtensionUrl = "https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-NHSCommunication"
+      * def commLanguageDls = response.extension.find(x => x.url == commExtensionUrl)
+      * def interpreterDls = commLanguageDls[1]
+      * def commLanguageindex = response.extension.findIndex(x => x.url == commExtensionUrl)
+      * def interpreterPath = "/extension/" + commLanguageindex + "/extension/1"
+      * def interpreter = response.extension[commLanguageindex].extension[1].valueBoolean
+      * def oppositeInterpreter = karate.eval('function getOppositeBoolean(value) { return !value; } getOppositeBoolean(' + karate.toString(interpreter) + ')')
+     
+    # Empty value on the patch
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * header Content-Type = "application/json-patch+json"
+      * header If-Match = karate.response.header('etag')
+      * path 'Patient', nhsNumber
+      * request 
+      """
+      {
+        "patches":[
+          {"op":"replace",
+          "path":"#(interpreterPath)",
+          "value": {
+                    "url": " ",
+                    "valueBoolean": "#(oppositeInterpreter)"              
+                   } 
+          }]}
+       """       
+      * method patch
+      * status 400
```

Comment:
I would be worth confirming that we are sending back the expected diagnostics message, which for this would be:
`The whole object must be sent for path '/extension' in Patient-Access mode`

Reply:
I think the error message will actually be
`This user does not have permission to update the fields in the patches provided.`

Reply:
Updated test to validate correct diagnostics message and status code is 400 for healthcare worker